### PR TITLE
feat: pnpm

### PR DESCRIPTION
**Pull request by wingkwong**
## Change Summary

This PR is to migrate npm to pnpm. However, it only applies to the existing packages that are in use currently, namely. `nocodb-sdk`, `nc-gui`, `nocodb` and `tests`. 

- closes: #5899
- ref: #6273
- update github action yaml files
- update docs
- remove unnecessary `package-lock.json`
- add pnpm workspace yaml
- bump some dependencies 
- remove `{ waitUntil: 'networkidle' }` from playwright
- return `goto` result from function since `load` needs to be fired after that
- bump to node 18
- Commands you need to know:
  - `pnpm bootstrap` (for installing all dependencies for sdk, frontend, backend including sdk build & playwright)
  - `pnpm --filter=<PKG> <CMD>` if you wanna execute a command in specific package.
  - `pnpm start:frontend`
  - `pnpm start:backend` 
- Before testing this PR, please remove `node_modules` and make sure you are using node 18.

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Executable 

```
curl http://dl.nocodb.com/0.111.0-pr-5903-20230831-0739/Noco-macos-arm64 -o noco -L \
&& chmod +x noco \
&& ./noco
```

### CODE REVIEW

**Comment by wingkwong** [comment]
🚨🚨🚨 ***Testing docker build. Do not merged!*** 🚨🚨🚨

File: packages/nc-gui/nuxt.config.ts
Comment (pranavxc) [inline packages/nc-gui/nuxt.config.ts:32]: The following meta is moved under app property in the latest version of Nuxt

```
 meta: {
    link: [
      {
        rel: 'icon',
        type: 'image/x-icon',
        href: './favicon.ico',
      },
    ],
    meta: [
      { charset: 'utf-8' },
      {
        name: 'viewport',
        content: 'width=device-width, initial-scale=1',
      },
      {
        hid: 'description',
        name: 'description',
        content: process.env.npm_package_description || '',
      },
    ],
  },
```

Comment (wingkwong) [reply packages/nc-gui/nuxt.config.ts:32]: moving to `app.head`.

File: .github/workflows/publish-dev-docs.yml
Comment (dammy001) [inline .github/workflows/publish-dev-docs.yml:20]: I think this should be specified in root package.json file so workflow can make use of it.

**Review by dammy001 — APPROVED** [review]

**Comment by wingkwong** [comment]
@pranavxc @dstala looks good now. let's have a final review.

**Review by dstala — APPROVED** [review]

**Comment by wingkwong** [comment]
Bad Timing

**Comment by OpenGG** [comment]
@wingkwong 
What is the current status of this PR?
Is support for PNPM still being considered?

**Comment by wingkwong** [comment]
@OpenGG Will be integrated again after #6156.

**Review by dstala — APPROVED** [review]

File: packages/nocodb/Dockerfile
Comment (pranavxc) [inline packages/nocodb/Dockerfile:45]: Can we use `pnpm exec modclean` here ?

Comment (wingkwong) [reply packages/nocodb/Dockerfile]: Good catch. However, I think it should be `pnpm dlx` instead since `modclean` is not in our dependencies.

**Review by pranavxc — COMMENTED** [review]
LGTM, great work @wingkwong  :+1:

Few things I want to confirm,

1. Any changes requires in `nocodb-seed` repo or does it break any existing node project?
2. Since we are migrating to pnpm docs requires update and hope it's already updated.
3. <strike>If user already development setup with nocodb repo, how to migrate to pnpm?</strike> - found answer from your pr description.

File: packages/nocodb/package.json
Comment (pranavxc) [inline packages/nocodb/package.json:131]: Always use exact version.

Comment (wingkwong) [reply packages/nocodb/package.json]: Updated. Later on we need to use `0.2.93` instead.

**Comment by wingkwong** [comment]
> 1. Any changes requires in `nocodb-seed` repo or does it break any existing node project?

I think you need to min node 18 in order to start. 

> 2. Since we are migrating to pnpm docs requires update and hope it's already updated.

The changes are already in this PR as well.

**Comment by wingkwong** [comment]
FYI. Adding `Do Not Merge` - `nc-help` needs to be synced.

**Comment by pranavxc** [comment]
Tested build commands(sdk, gui and backend) and works fine ✅

**Pull request merged**
